changelog/unreleased/pseudo-tx-share.md

```diff
@@ -0,0 +1,8 @@
+Enhancement: pseudo-transactionalize sharing
+
+Currently, sharing is not transactionalized: setting ACLs and writing the share to the db is completely independent. Because of this, sometimes a share is written to the database even though the ACLs have not been set. This enhancement fixes this by
```

Comment:
You do not fix it; you make the situation better. Just explain that the DB falsely led users to believe it was shared and is more reliable than the storage, so we improve the situation by doing the least reliable first.